### DISCUSSION TITLE
Preactions

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#183] [Ideas]
# Background

I have planned to create a discussion for this a few months, but I've always forgotten it. Then again, this topic will not contain very much stuff that hasn't been discussed elsewhere. The feature itself has been mentioned in other discussions, and this particular discussion is now just about terminology.

This feature has been partly covered in the following discussions:
- https://github.com/Taitava/obsidian-shellcommands/discussions/17#discussioncomment-1454926
- https://github.com/Taitava/obsidian-shellcommands/discussions/54#discussioncomment-1451722
- https://github.com/Taitava/obsidian-shellcommands/discussions/60#discussioncomment-1790237

# On to the actual topic: What are preactions/preparations

`0.12.0` will have a feature named _Preactions_ or _Preparations_. Basically, it's a _list of actions that will be performed **before** executing a shell command_. Preactions can be defined separately for each shell command.

For example, the current _Confirm before execution_ feature, that has been around since [`0.4.0`](https://github.com/Taitava/obsidian-shellcommands/issues/35), can be thought as a preaction/preparation. It has not been referred to as a _preaction_ or _preparation_ yet, but in `0.12.0` it will be counted as one. It prepares the execution by asking from the user, should the shell command really be executed, or cancelled.

Lately, I have talked more about [a modal for asking values from user (Prompt)](https://github.com/Taitava/obsidian-shellcommands/discussions/17). These _Prompts_ are part of _Preactions_/_Preparations_. A prompt will open up just before a shell command is executed. A prompt prepares some of the variables used by the shell command.

A third preaction/preparation will be #54 sometime in the future. It will _prepare_ the shell command execution by selecting some text in the current file, if nothing is selected at the moment of execution.

# Which term should be used?

So far, I have named the feature to be _Preactions_. But that sounds a bit bad English to me, but maybe I'm just not jused to use that word. I might change the name to _Preparations_.

@FelipeRearden what do you think? 🙂

**Comment by FelipeRearden** [comment]
Hello @Taitava !!!!!!

I like `Preaction` and I have some ideas to share ...

- Precommand
- Starter
- Starter Action  
- Introduction
- Epilogue 
- Introaction 😜 

The good thing about Preaction is that Action is a world that everybody in the world knows (like Love, apple and God). 

Have a great day!!!!

**Reply by Taitava** [reply]
_Preactions_ it is, then! 🙂👍

Have a nice day, too! 😎
